cli/src/lib.rs

```diff
@@ -86,8 +86,8 @@ struct ValidationWokerCommand {
 pub fn run<W>(worker: W, version: cli::VersionInfo) -> error::Result<()> where
 	W: Worker,
 {
-	let command = cli::parse_and_execute::<service::Factory, PolkadotSubCommands, NoCustom, _, _, _, _, _>(
-		load_spec, &version, "parity-polkadot", std::env::args(), worker,
+	match cli::parse_and_prepare::<PolkadotSubCommands, NoCustom, _>(&version, "parity-polkadot", std::env::args()) {
+		cli::ParseAndPrepare::Run(cmd) => cmd.run(load_spec, worker,
 		|worker, _cli_args, _custom_args, mut config| {
```

Comment:
wouldn't we want to have this version information on all subcommands?

Reply:
if this is an issue, let's do it in a followup PR